e2e_playwright/st_experimental_fragment_basics_test.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import wait_for_app_run
+
+
+def get_uuids(app: Page):
+    text_in_fragment = app.get_by_test_id("stMarkdown").first.text_content()
+    text_outside_fragment = app.get_by_test_id("stMarkdown").last.text_content()
+
+    return text_in_fragment, text_outside_fragment
+
+
+def assert_only_fragment_uuid_changed(
+    app: Page, old_text_in_fragment: str, old_text_outside_fragment: str
+):
+    new_text_in_fragment, new_text_outside_fragment = get_uuids(app)
+    assert old_text_in_fragment != new_text_in_fragment
+    assert old_text_outside_fragment == new_text_outside_fragment
```

Comment:
The usage of assert here is almost guaranteed to cause flakiness :( See playwright rule 1 and 3 [here](https://github.com/streamlit/streamlit/wiki/Running-e2e-tests-and-updating-snapshots#three-rules-of-playwright) 😅

Either use except, e.g.: 

```suggestion
    except(app.get_by_test_id("stMarkdown").first).not_to_have_text(old_text_in_fragment)
    except(app.get_by_test_id("stMarkdown").last).not_to_have_text(old_text_outside_fragment)
```

Or use the `wait_until` method from conftest.

Reply:
👍 swapped to using `expect(...).not_to_have_text(...)`